File: OOPs/abstraction.cpp
class animal{
    private:
        string name;
    public:
        void setName(string n){
            name=n;
        }
        virtual string getName() = 0 // pure virtual function has to be overridden in derived classes
}

class dog : public animal{
    public:
        string getName() override{
            return "Dog";
        }
}

File: OOPs/abstraction.js
class animal{
    constructor(name){
        this._name = name;
    }
    setName(name){
        this._name=name;
    }
    getName(){
        throw new Error("It is to be implemented in derived classes");
    }
}

class dog extends animal{
    getName(){
        return "Dog";
    }
}